### ISSUE TITLE
💣 Calling suggest causes NSRangeException

### ISSUE THREAD

**Issue by andreyz** [helpscout/beacon-ios-sdk#102]
* Beacon install method: [ ] Carthage, [ ] CocoaPods, [x] Manual

**Actual outcome**
Following call sequence causes the app to crash:
1. `HSBeacon.navigate("/ask/message/", beaconSettings: __ :viewController:__)`
2. Dismiss Beacon view.
3. Call `HSBeacon.suggest` with array of 3 article IDs, followed by `HSBeacon.open(settings, from: viewController)`
4. BOOM! 💣 

It must be something related to (pre-)loading, as a different sequence of method calls doesn't case a crash.

```
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray'
```
```
#0	0x000000018e4e2a48 in __exceptionPreprocess ()
#1	0x000000018e209fa4 in objc_exception_throw ()
#2	0x000000018e3e2268 in -[__NSArray0 objectAtIndex:] ()
#3	0x0000000104001bd8 in -[HSBeacon setCustomSuggestions] at /Users/travis/build/helpscout/beacon-ios-sdk-private/Beacon/Classes/HSBeacon.m:239
#4	0x000000010400091c in __69-[HSBeacon openBeacon:fromViewController:signature:postLoadCallback:]_block_invoke.78 at /Users/travis/build/helpscout/beacon-ios-sdk-private/Beacon/Classes/HSBeacon.m:136
#5	0x00000001059817fc in _dispatch_call_block_and_release ()
#6	0x0000000105982bd8 in _dispatch_client_callout ()
#7	0x0000000105990c34 in _dispatch_main_queue_callback_4CF ()
#8	0x000000018e4605e4 in __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ ()
#9	0x000000018e45b5d8 in __CFRunLoopRun ()
#10	0x000000018e45aadc in CFRunLoopRunSpecific ()
#11	0x00000001983e0328 in GSEventRunModal ()
#12	0x0000000192555ae0 in UIApplicationMain ()
#13	0x000000010314e228 in main at <REDACTED>.swift:12
#14	0x000000018e2e4360 in start ()
```

**Expected outcome**
1. Beacon should present document viewer with suggested articles displayed.
2. Beacon mobile team using a suite of UI tests that catch this kind of issues before the SDK ships.

**Comment by codenameDuchess** [comment]
Hi @andreyz, thanks for this detailed bug report. We'll circle back and let you know once the release with the fix is live.

Internal tracking link(https://helpscout.atlassian.net/browse/IOSSDK-338).

**Comment by ScottPetit** [comment]
Hey 👋 @andreyz,

Thanks for reporting this, just wanted to write and let you know that this is resolved with the latest version (2.0.2) of the SDK.  If you'd like to see what else is in this release take a look at the [Changelog](https://github.com/helpscout/beacon-ios-sdk/blob/master/CHANGELOG.md).
